### PR TITLE
Make android-extensions plugin always produce compilation error

### PR DESCRIPTION
**Pull request by madsager**

### CODE REVIEW

**Comment by madsager** [comment]
@udalov This will make the plugin unusable. However, all IDE code that depends on android-extensions should still work, and all the tests of the code still pass (after these changes). This looks like the minimal change we can do for 1.8 for deprecation. I'll then follow up with more deprecation work once 1.8 has been branched. What do you think?

File: libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/android/internal/AndroidSubplugin.kt [hunk 1]
```diff
@@ -36,8 +36,8 @@ class AndroidExtensionsSubpluginIndicator @Inject internal constructor(private v
         registry.register(KotlinAndroidExtensionModelBuilder())
         project.plugins.apply(AndroidSubplugin::class.java)
 
-        project.logger.warn(
-            "Warning: The 'kotlin-android-extensions' Gradle plugin is deprecated. " +
+        project.logger.error(
+            "Error: The 'kotlin-android-extensions' Gradle plugin is deprecated. " +
```
[the remark this answers is not in the document]
Comment (madsager) [reply libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/android/internal/AndroidSubplugin.kt]: Done.

**Comment by udalov** [comment]
CLI tests in `CliTestGenerated$Plugins` should use some other plugin I guess:
* testPluginSimple_legacy
* testPluginWithK2Error_legacy
* testPluginWithK2Error_modern

**Comment by madsager** [comment]
Yes, sorry, I should have anticipated that. I'll get going on updating this. The CLI tests should be removed as they explicitly test synthetics. The Android tests should be rewritten (I have that code in the other larger change) or be removed when they only test synthetics.

**Comment by madsager** [comment]
Android test projects updated, the ones that test android-extensions only removed. The CLI tests that use android-extensions removed. There are other CLI tests that use other plugins.

**Comment by udalov** [comment]
Thank you!

**Pull request merged**